packages/flutter/lib/src/widgets/container.dart

```diff
@@ -388,47 +382,47 @@ class Container extends StatelessWidget {
     }
 
     if (alignment != null)
-      current = Align(alignment: alignment, child: current);
+      current = Align(alignment: alignment!, child: current);
 
-    final EdgeInsetsGeometry effectivePadding = _paddingIncludingDecoration;
+    final EdgeInsetsGeometry? effectivePadding = _paddingIncludingDecoration;
     if (effectivePadding != null)
       current = Padding(padding: effectivePadding, child: current);
 
     if (color != null)
-      current = ColoredBox(color: color, child: current);
+      current = ColoredBox(color: color!, child: current);
 
     if (decoration != null)
-      current = DecoratedBox(decoration: decoration, child: current);
+      current = DecoratedBox(decoration: decoration!, child: current);
 
     if (foregroundDecoration != null) {
       current = DecoratedBox(
-        decoration: foregroundDecoration,
+        decoration: foregroundDecoration!,
         position: DecorationPosition.foreground,
         child: current,
       );
     }
 
     if (constraints != null)
-      current = ConstrainedBox(constraints: constraints, child: current);
+      current = ConstrainedBox(constraints: constraints!, child: current);
 
     if (margin != null)
-      current = Padding(padding: margin, child: current);
+      current = Padding(padding: margin!, child: current);
 
     if (transform != null)
-      current = Transform(transform: transform, child: current);
+      current = Transform(transform: transform!, child: current);
 
     if (clipBehavior != Clip.none) {
```

Comment:
add a `decoration != null` here to avoid the null deref below

Reply:
Done